File: app/src/main/java/com/example/jacob/myapplication/DeepAnalysis/Word.java
package com.example.jacob.myapplication.DeepAnalysis;

import java.util.Objects;

/**
 * Created by jacob on 25/11/2015.
 */
public class Word extends SentimentData {
    public String name;
    boolean swearWord;
    boolean emoticon;

    public Word(String name, float aggresiveness, float fear, float happiness, float love, float sadness, boolean emoticon,  boolean swearWord) {
        super(aggresiveness, fear, happiness, love, sadness);
        this.emoticon = emoticon;
        this.name = name;
        this.swearWord = swearWord;
    }

    @Override
    public boolean equals(Object o) {
        if (o.getClass() == String.class)
            return name.equals(o);

        if (o.getClass() == Word.class)
            return ((Word) o).name.equals(this.name);

        return false;
    }
}

File: app/src/main/java/me/jacobrr/whatsappanalyzer/analysis/Word.kt
package me.jacobrr.whatsappanalyzer.analysis

/**
 * Created by jacob on 25/11/2015.
 */
class Word(var name: String, aggressiveness: Float, fear: Float, happiness: Float, love: Float, sadness: Float, internal var emoticon: Boolean, internal var swearWord: Boolean) : SentimentData(aggressiveness, fear, happiness, love, sadness) {

    override fun equals(other: Any?): Boolean {
        if (other!!.javaClass == String::class.java)
            return name == other

        if (other.javaClass == Word::class.java)
            return (other as Word).name == this.name

        return false
    }

    override fun hashCode(): Int {
        return name.hashCode()
    }
}
